### PR TITLE
Fix concurrency issues in FakeValuesService

### PR DESCRIPTION
**Pull request by rwong-gw**
* Replace  IdentityHashMaps with ConcurrentHashMaps
* Replace WeakHashMaps with Guava weak-key caches
* Add dependency for Guava 31.1-jre

### CODE REVIEW

File: src/main/java/net/datafaker/service/FakeValuesService.java [hunk 1]
```diff
@@ -1,5 +1,7 @@
 package net.datafaker.service;
 
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
```
Comment (kingthorin) [inline src/main/java/net/datafaker/service/FakeValuesService.java:4]: I think these can go?

File: src/main/java/net/datafaker/service/FakeValuesService.java [hunk 2]
```diff
@@ -22,17 +24,8 @@
...
-import java.util.Collection;
-import java.util.Collections;
-import java.util.HashMap;
-import java.util.IdentityHashMap;
-import java.util.List;
-import java.util.Locale;
-import java.util.Map;
-import java.util.Objects;
-import java.util.WeakHashMap;
+import java.util.*;
```
Comment (kingthorin) [inline src/main/java/net/datafaker/service/FakeValuesService.java:27]: Specific imports should be used vs wildcard

File: src/main/java/net/datafaker/service/FakeValuesService.java [hunk 3]
```diff
@@ -1,19 +1,9 @@
...
-import net.datafaker.providers.base.BaseFaker;
-import net.datafaker.providers.base.Name;
-import net.datafaker.providers.base.ProviderRegistration;
-import net.datafaker.transformations.CsvTransformer;
-import net.datafaker.transformations.Field;
-import net.datafaker.transformations.JsonTransformer;
-import net.datafaker.transformations.Schema;
-import net.datafaker.transformations.SimpleField;
+import net.datafaker.providers.base.*;
+import net.datafaker.transformations.*;
```
Comment (kingthorin) [inline src/main/java/net/datafaker/service/FakeValuesService.java:6]: 🤦‍♂️

**Review by snuyanzin — CHANGES_REQUESTED** [review]
I'm a bit skeptic about this change without reproducible unit test.

Moreover replacement of e.g. IdentityHashMaps with ConcurrentHashMaps leads to some slow downs and this should be kept in mind

File: src/main/java/net/datafaker/service/FakeValuesService.java [hunk 4]
```diff
@@ -222,27 +219,7 @@ private String[] split(String string) {
...
-        for (int i = 0; i < length; i++) {
-            if (string.charAt(i) == splitChar) {
-                if (i - start > 0) {
-                    result[j++] = String.valueOf(chars, start, i - start);
-                }
-                start = i + 1;
-            }
-        }
-        result[j] = String.valueOf(chars, start, chars.length - start);
+        result = string.split("\\.");   // String#split is plenty efficient for single-char patterns
```
Comment (snuyanzin) [inline src/main/java/net/datafaker/service/FakeValuesService.java:222]: This statement does not look like true

please provide proof for that.

Dummy test with jmh shows that current code is about 10-15% faster than `result = string.split("\\.");`

```
Benchmark             Mode  Cnt      Score      Error   Units
Issue760.testSplit   thrpt    5  24329.658 ±  734.929  ops/ms
Issue760.testSplit2  thrpt    5  21338.631 ± 1043.678  ops/ms
```
code snippet
```java
@BenchmarkMode(Mode.AverageTime)
@OutputTimeUnit(TimeUnit.MILLISECONDS)
@State(Scope.Benchmark)
@Fork(value = 2, jvmArgs = {"-Xms2G", "-Xmx2G"})
public class Issue760 {
    private static final String STRING_2_SPLIT = "Address.streetAddress";
    public static void main(String[] args) throws RunnerException {

        Options opt = new OptionsBuilder()
                .include(Issue760.class.getSimpleName())
                .forks(1)
                .build();

        new Runner(opt).run();
    }

    @Benchmark
    @BenchmarkMode(Mode.Throughput)
    public void testSplit(Blackhole blackhole) {
        final String string = STRING_2_SPLIT;
        int size = 0;
        final char splitChar = '.';
        final int length = string.length();
        for (int i = 0; i < length; i++) {
            if (string.charAt(i) == splitChar) {
                size++;
            }
        }
        String[] result = new String[size + 1];
        final char[] chars = string.toCharArray();
        int start = 0;
        int j = 0;
        for (int i = 0; i < length; i++) {
            if (string.charAt(i) == splitChar) {
                if (i - start > 0) {
                    result[j++] = String.valueOf(chars, start, i - start);
                }
                start = i + 1;
            }
        }
        result[j] = String.valueOf(chars, start, chars.length - start);
        blackhole.consume(result);
    }

    @Benchmark
    @BenchmarkMode(Mode.Throughput)
    public void testSplit2(Blackhole blackhole) {
        final String string = STRING_2_SPLIT;
        String[] result = string.split("\\.");
        blackhole.consume(result);
    }
}

```

**Comment by snuyanzin** [comment]
Also see my comment https://github.com/datafaker-net/datafaker/issues/759#issuecomment-1505644458

**Comment by snuyanzin** [comment]
Closing in favor of #770

**Pull request closed without merging**